lib/grammar.pegjs

```diff
@@ -145,6 +149,26 @@ param_direction = 'in' __ 'req' { return 'in req'; } / 'in' __ 'opt' { return 'i
     'out' { return 'out'; }
 
 
+// Dataset & Thingpedia Definition
+dataset_name = ident / name:class_name { return '@' + name; }
+
+dataset = _ 'dataset' _ name:dataset_name _ lan:('language' _ literal_string _)? '{' _ examples:(_ example _)* _ '}' {
+    return new Ast.Dataset(name, lan? lan[2] : 'en' , take(examples, 1));
+}
+example = 'stream'  _ params:(lambda_param_decl_list)? _ ':=' _ stream:stream _ annotation:nl_annotation _ ';' {
+    if (annotation.name !== 'utterances') throw new Error('Only "utterances" annotation is allowed for examples');
+    return new Ast.Example('stream', params ? params : [], stream, annotation.value.value);
+} / 'query'  _ params:(lambda_param_decl_list)? _ ':=' _ query:table _ annotation:nl_annotation _ ';' {
+    if (annotation.name !== 'utterances') throw new Error('Only "utterances" annotation is allowed for examples');
+    return new Ast.Example('query', params ? params : [], query, annotation.value.value);
+} / 'action'  _ params:(lambda_param_decl_list)? _ ':=' _ action:action _ annotation:nl_annotation _ ';' {
+    if (annotation.name !== 'utterances') throw new Error('Only "utterances" annotation is allowed for examples');
+    return new Ast.Example('action', params ? params : [], action, annotation.value.value);
+} / 'program'  _ params:(lambda_param_decl_list)? _ ':=' _ program:program _ annotation:nl_annotation _ ';' {
+    if (annotation.name !== 'utterances') throw new Error('Only "utterances" annotation is allowed for examples');
+    return new Ast.Example('program', params ? params : [], program, annotation.value.value);
+}
```

Comment:
Do we not allow repeated annotations? I thought we agreed `#_[ann="one"] #_[ann="two"]` was equivalent to `#_[ann=["one", "two"]]`? Or not?

Reply:
I think this is just a design choice. I would like to keep the syntax I have unless there are some benefits to make the change.